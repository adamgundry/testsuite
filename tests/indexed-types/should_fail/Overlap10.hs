{-# LANGUAGE TypeFamilies #-}

module Overlap10 where

type family F a b
type instance where
  F a a = Int
  F a b = b

g :: F a Bool
g = False




