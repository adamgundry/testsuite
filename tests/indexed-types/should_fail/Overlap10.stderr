
Overlap10.hs:11:5:
    Couldn't match expected type `F a Bool' with actual type `Bool'
    Relevant bindings include
      g :: F a Bool (bound at Overlap10.hs:11:1)
    In the expression: False
    In an equation for `g': g = False
